trino-lb/src/maintenance/query_count_fetcher.rs

```diff
@@ -152,8 +148,34 @@ impl QueryCountFetcher {
         }
     }
 
-    #[instrument(skip(self))]
-    async fn process_cluster(&self, cluster: &TrinoClusterConfig) {
+    #[instrument(skip(self, cluster), fields(cluster_name = cluster.name))]
+    async fn process_cluster(&self, cluster: &TrinoClusterConfig, state: ClusterState) {
+        match state {
+            ClusterState::Ready | ClusterState::Unhealthy | ClusterState::Draining { .. } => {
+                self.fetch_and_store_query_count(cluster).await;
+            }
+            ClusterState::Unknown
+            | ClusterState::Stopped
+            | ClusterState::Starting
+            | ClusterState::Terminating
+            | ClusterState::Deactivated => {
+                if let Err(err) = self
+                    .persistence
+                    .set_cluster_query_count(&cluster.name, 0)
+                    .await
```

Comment:
question: Why exactly do we set the number of queries to zero here? Can you add a developer comment here which shortly explains it?

Reply:
I added some rustdoc in 373ab07 explaining it. Is that sufficient in this case here?